frontend/packages/kubevirt-plugin/src/selectors/service/selectors.ts

```diff
@@ -1,7 +1,24 @@
 import * as _ from 'lodash';
 import { K8sResourceKind } from '@console/internal/module/k8s';
+import { ServiceKind } from '@console/knative-plugin/src/types';
+import { VMIKind } from '../../types';
+import { getVmiTemplateLabels } from '../vmi/basic';
 
 export const getServicePort = (service: K8sResourceKind, targetPort: number) =>
   _.get(service, ['spec', 'ports'], []).find(
     (servicePort) => targetPort === servicePort.targetPort,
   );
+
+const getServiceSelectors = (service: ServiceKind) =>
+  service && service.spec && service.spec.selector ? service.spec.selector : {};
```

Comment:
we could use lodash also here

Reply:
The drawback of this approach is performance. I do not have any numbers for that but instead of simple direct access with a few conditions we get a function call plus string parsing. On the other hand, these three operands fit onto a single line and there is lower probability of making a typo there (thanks to typescript).

I would focus on https://github.com/tc39/proposal-optional-chaining, once ready.

Reply:
it is not as bad as catching exceptions. I would still use it when checking for longer paths